--- a/lib/RT/Client/REST.pm	2013-09-04 11:06:56.000000000 +0200
+++ b/lib/RT/Client/REST.pm	2013-11-05 18:06:53.885106887 +0100
@@ -155,7 +155,7 @@
     my $id = $self->_valid_numeric_object_id(delete($opts{id}));
 
     my $form = form_parse(
-        $self->_submit("$type/$parent_id/attachments/$id")->decoded_content
+        $self->_submit("$type/$parent_id/attachments/$id")->content
     );
     my ($c, $o, $k, $e) = @{$$form[0]};
 
@@ -518,7 +518,7 @@
     # Now, we construct the request.
     if (@$data) {
         # The request object expects "bytes", not strings
-        map { utf8::encode($_) } @$data;
+        map { utf8::encode($_) unless ref($_)} @$data;
 
         $req = POST($self->_uri($uri), $data, Content_Type => 'form-data');
     }
